artifactory/services/users.go

```diff
@@ -0,0 +1,97 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"net/http"
+)
+
+// application/vnd.org.jfrog.artifactory.security.User+json
+type User struct {
+	Name                     string   `json:"name,omitempty"`
+	Email                    string   `json:"email,omitempty"`
+	Password                 string   `json:"password,omitempty"`
+	Admin                    bool     `json:"admin,omitempty"`
+	ProfileUpdatable         bool     `json:"profileUpdatable,omitempty"`
+	DisableUIAccess          bool     `json:"disableUIAccess,omitempty"`
+	InternalPasswordDisabled bool     `json:"internalPasswordDisabled,omitempty"`
+	LastLoggedIn             string   `json:"lastLoggedIn,omitempty"`
+	Realm                    string   `json:"realm,omitempty"`
+	Groups                   []string `json:"groups,omitempty"`
+}
+
+type UserService struct {
+	client     *jfroghttpclient.JfrogHttpClient
+	ArtDetails auth.ServiceDetails
+}
+
+func NewUserService(client *jfroghttpclient.JfrogHttpClient) *UserService {
+	return &UserService{client: client}
+}
+
+func (us *UserService) SetArtifactoryDetails(rt auth.ServiceDetails) {
+	us.ArtDetails = rt
+}
+
+func (us *UserService) GetUser(name string) (*User, error) {
+	httpDetails := us.ArtDetails.CreateHttpClientDetails()
+	url := fmt.Sprintf("%sapi/security/users/%s", us.ArtDetails.GetUrl(), name)
+	res, body, _, err := us.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode > http.StatusNoContent {
```

Comment:
As discussed earlier, let's avoid checking http response code ranges.Instead, let's check the "good" response code, otherwise, return return.
This applies to other places in the code as well.